docs/v3/core/index.rst

```diff
@@ -1598,38 +1520,125 @@ Storage transformers may be stacked to combine different functionalities:
 
 If such extension points are used by groups or arrays, they are required.
 
-See https://github.com/zarr-developers/zarr-specs/issues/49 for a list of
-potential extensions.
+New extension points may be proposed to the Zarr community through the ZEP
+process. See `ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ for more information.
+
+Extension definition
+--------------------
+
+Extensions are defined in `metadata documents`_ either as objects or as
+short-hand names. If using an objection definition, the following pattern
+MUST be followed::
+
+    {
+        "name": "<name>",
+        "configuration": { ... } # optional
+    }
+
+If such an object is present, the field `must_understand` is implicitly set to
+`True` and an object can explicitly set `must_understand=False` if
+implementations can ignore its presence, following the current guidelines in
+the v3 specification.
+
+Instead of extension objects, short-hand names may continue to be used if no
+configuration metadata is required. They would be equivalent to extension
+objects with just a `name` key.
 
-Specifications for new extensions are recommended to be published in the
-https://github.com/zarr-developers/zarr-specs repository via the
-`ZEP process <https://zarr.dev/zeps/active/ZEP0000.html>`_. If a specification
-is published decentralized (e.g. for initial experimentation or due to a very
-specialized scope), it must use a URL in the `name` key of its metadata, which
-identifies the publishing organization or individual, and should point to the
-specification of the extension.
+Extension naming
+----------------
+
+The identifier used in the `name` field of the extension definition can follow one of two forms:
+
+1. **Raw names** MUST be assigned within a central repository and follow the
+   compatibility and versioning v3 `stability policy`_.
+   The name assignment is managed through the `zarr-extensions`_
+   Github repository, where each extension is
+   listed and either contains a spec document or links to a spec document.
+   Names are never unassigned or reassigned. The Zarr Steering Council or by delegation a
+   maintainer team reserves the right to refuse name assignment at its own
+   discretion.
+
+   - **Example:** ``zstd``
+   - **Acceptd regex:** ``^[a-z0-9-_.]+$``
+
+2. **URI-based names** can be used by anyone without further coordination
+   though the assumption is that users reasonably "own" the URI. The URL SHOULD
+   resolve to a human-readable explanation of the extension, but
+   implementations SHOULD NOT attempt to resolve the URL during processing.
+   There are no guarantees in terms of versioning or compatibility. However,
+   preserving backwards-compatibility is strongly encouraged. See the
+   [versioning section](#Versioning-and-spec-evolution) below.
+
+   - **Example:** ``https://example.com/zarr3/consolidated-metadata``
+   - **Accepted regex:** ``^https?:\/\/[^/?#]+[^?#]*$``
```

Comment:
why is https required?

Reply:
It is `http` or `https`. The `?` in the regex makes the `s` optional.

Reply:
so then why is http or https required?